rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/edition/EditDynamicParametersWindow.java

```diff
@@ -96,56 +93,39 @@ protected void modifyFormItemsAfterCreation() {
+                                                   .collect(Collectors.toList());
+
+        for (FormItem formItem : this.nodeSourcePluginsForm.getFields()) {
+            if (isNotDynamic(formItem, dynamicParametersName) &&
+                isFocusedPluginItem(focusedInfrastructurePlugin, focusedPolicyPlugin, formItem)) {
+                formItem.disable();
             }
         }
     }
 
-    private boolean isItemEqualToDynamicField(FormItem formItem, String fieldFullName) {
-        return formItem.getName().equals(fieldFullName) ||
-               formItem.getName().equals(fieldFullName + EDIT_FORM_ITEM_SUFFIX);
+    private Stream<String> getDynamicParametersName(PluginDescriptor focusedInfrastructurePlugin) {
+        return focusedInfrastructurePlugin.getConfigurableFields()
+                                          .stream()
+                                          .filter(PluginDescriptor.Field::isDynamic)
+                                          .map(field -> focusedInfrastructurePlugin.getPluginName() + field.getName());
     }
 
-    private void filterAndDisableNonDynamicItem(FormItem formItem) {
-        if (this.formItemsByName.keySet().stream().anyMatch(pluginName -> formItem.getName().startsWith(pluginName)) ||
-            formItem.getName().equals(INFRASTRUCTURE_FORM_KEY) || formItem.getName().equals(POLICY_FORM_KEY)) {
-            disableNonDynamicItem(formItem);
-        }
+    private boolean isFocusedPluginItem(PluginDescriptor focusedInfrastructurePlugin,
+            PluginDescriptor focusedPolicyPlugin, FormItem formItem) {
+        return formItem.getName().startsWith(focusedInfrastructurePlugin.getPluginName()) ||
+               formItem.getName().startsWith(focusedPolicyPlugin.getPluginName()) ||
+               formItem.getName().equals(INFRASTRUCTURE_FORM_KEY) || formItem.getName().equals(POLICY_FORM_KEY);
     }
 
-    private void disableNonDynamicItem(FormItem formItem) {
-        formItem.disable();
-        // when a form item is disabled, it will not be given as parameter of
-        // the submitted form, so we need to add the item content in a hidden
-        // item for it to be submitted with the form. This is true for all
-        // form items but the SelectItem (they are submitted as part of the
-        // form even if disabled), this is why we make a special case.
-        if (!(formItem instanceof SelectItem)) {
-            HiddenItem hiddenItem = new HiddenItem(formItem.getName());
-            hiddenItem.setValue(formItem.getValue());
-            this.formItemsByName.put(formItem.getName(), Collections.singletonList(hiddenItem));
-        }
+    private boolean isNotDynamic(FormItem formItem, List<String> dynamicParametersName) {
+        return dynamicParametersName.stream().noneMatch(fieldFullName -> isDynamic(formItem, fieldFullName));
+    }
+
+    private boolean isDynamic(FormItem formItem, String fieldFullName) {
```

Comment:
if form name equals to full field name then something is dynamic. maybe you could rename parameters.

Reply:
Yes you are right, renamed.